### PR TITLE
Create endpoint that receives day-time, hourly and night-time metrics from AWS IoT

### PR DESCRIPTION
**Pull request by NLSanyu**
This PR creates an endpoint that receives day-time, hourly and night-time metrics from AWS IoT, then aggregates the data and stores it in the DailyAggregate and HourlyAggregate model tables

### How to test:

Send an POST API request to *.../analysis/receive_iot_data/*, with a request body formatted as follows:
```
{
    "type": "hourly",
    "data": [...the list of records]
}
```

### Example:
Screenshot 2022-11-10 at 18 42 55

### CODE REVIEW

**Comment by NLSanyu** [comment]
@cmplx-xyttmt this code might need some refactoring (I have a million ideas on how to refactor and optimize it), but for now, it does the task as it is supposed to 😆

File: analysis/models.py [hunk 1]
```diff
@@ -17,8 +17,7 @@ class HourlyAggregate(models.Model):
     hourly_median_db_level = models.FloatField(null=True)
     hourly_max_db_level = models.FloatField(null=True)
     hourly_no_of_exceedances = models.PositiveIntegerField(null=True)
-    device_name = models.CharField(max_length=255, null=True)
-    # device = models.OneToOneField(Device, on_delete=models.CASCADE)
+    device = models.ForeignKey(Device, on_delete=models.CASCADE)
 
 
 class DailyAggregate(models.Model):
```
Comment (cmplx-xyttmt) [inline analysis/models.py:23]: @NLSanyu, I think there should also be a field here that says whether it's daytime or nighttime aggregate

**Review by cmplx-xyttmt — CHANGES_REQUESTED** [review]
Requested 1 change

File: analysis/models.py (hunk 1 above)
Comment (NLSanyu) [reply analysis/models.py:23]: This is going to sound strange, but I've been struggling for 15 minutes over what name to give this field 😅

Comment (NLSanyu) [reply analysis/models.py:23]: > @NLSanyu, I think there should also be a field here that says whether it's daytime or nighttime aggregate

This is going to sound strange, but I've been struggling for 15 minutes over what name to give this field 😅

**Review by cmplx-xyttmt — APPROVED** [review]

**Pull request merged**